src/renderer/index.tsx

```diff
@@ -41,74 +37,9 @@ const ImageOptimizer = (props: IProps) => {
 		[siteID],
 	);
 
-	// listen for optimization events and update status accordingly
-	useEffect(
-		() => {
-			if (!ipcRenderer.listenerCount(IPC_EVENTS.COMPRESS_IMAGE_SUCCESS)) {
-				ipcRenderer.on(
-					IPC_EVENTS.COMPRESS_IMAGE_SUCCESS,
-					(_, imageData: ImageData) => {
-						store.dispatch(actions.optimizeSuccess({ siteID, imageData }));
-					},
-				);
-			}
-
-			if (!ipcRenderer.listenerCount(IPC_EVENTS.COMPRESS_IMAGE_FAIL)) {
-				ipcRenderer.on(
-					IPC_EVENTS.COMPRESS_IMAGE_FAIL,
-					(_, imageID, errorMessage) => {
-						store.dispatch(actions.optimizeFailure({ siteID, imageID, errorMessage }));
-					},
-				);
-			}
-
-			if (!ipcRenderer.listenerCount(IPC_EVENTS.COMPRESS_IMAGE_STARTED)) {
-				ipcRenderer.on(
-					IPC_EVENTS.COMPRESS_IMAGE_STARTED,
-					(_, imageID) => {
-						store.dispatch(actions.optimizationStarted({ siteID, imageID }));
-					},
-				);
-			}
-
-			if (!ipcRenderer.listenerCount(IPC_EVENTS.COMPRESS_ALL_IMAGES_COMPLETE)) {
-				ipcRenderer.on(
-					IPC_EVENTS.COMPRESS_ALL_IMAGES_COMPLETE,
-					async () => {
-						store.dispatch(actions.optimizationComplete({ siteID }));
-					},
-				);
-			}
-
-			if (!ipcRenderer.listenerCount(IPC_EVENTS.SCAN_IMAGES_COMPLETE)) {
-				ipcRenderer.on(
-					IPC_EVENTS.SCAN_IMAGES_COMPLETE,
-					(_, siteImageData) => {
-						store.dispatch(actions.scanSuccess({ siteID, siteImageData }));
-					},
-				);
-			}
-
-			if (!ipcRenderer.listenerCount(IPC_EVENTS.SCAN_IMAGES_FAILURE)) {
-				ipcRenderer.on(
-					IPC_EVENTS.SCAN_IMAGES_FAILURE,
-					(_, error) => {
-						store.dispatch(actions.scanFailure({ siteID, error }));
-					},
-				);
-			}
-
-			return () => {
-				ipcRenderer.removeAllListeners(IPC_EVENTS.COMPRESS_IMAGE_STARTED);
-				ipcRenderer.removeAllListeners(IPC_EVENTS.COMPRESS_IMAGE_FAIL);
-				ipcRenderer.removeAllListeners(IPC_EVENTS.COMPRESS_IMAGE_SUCCESS);
-				ipcRenderer.removeAllListeners(IPC_EVENTS.COMPRESS_ALL_IMAGES_COMPLETE);
-				ipcRenderer.removeAllListeners(IPC_EVENTS.SCAN_IMAGES_COMPLETE);
-				ipcRenderer.removeAllListeners(IPC_EVENTS.SCAN_IMAGES_FAILURE);
-			}
-		},
-		[siteImageData],
-	);
+	const setOverviewSelected = (overviewSelected: boolean) => {
+		store.dispatch(actions.overviewSelected({ siteID, overviewSelected }));
+	}
```

Comment:
As far as I can tell, this is the state object used to persist whether a scan is running or not, correct?

Reply:
Correct!

Reply:
Added a field to the `SiteImageData` type: https://github.com/getflywheel/local-addon-image-optimizer/pull/48/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR29